app/src/main/java/com/infomaniak/mail/data/models/thread/Thread.kt

```diff
@@ -107,4 +124,23 @@ class Thread : RealmObject {
         STARRED("Starred"),
         UNSTARRED("Unstarred"),
     }
+
+    companion object {
+        fun from(message: Message, threadUid: String?) = Thread().apply {
+            uid = threadUid ?: message.uid
+            from = message.from
+            to = message.to
+            bcc = message.bcc
+            cc = message.cc
+            hasDrafts = true
+            hasAttachments = message.hasAttachments
+            messages = realmListOf(message)
+            subject = message.subject
+            date = message.date
+            messagesCount = 1
+            size = message.attachments.size
+            flagged = message.flagged
+            displayedDate = message.date?.toDate()?.formatDate() ?: ""
+        }
+    }
```

Comment:
to see if we can simplify it with an interface

Reply:
It will be done later in a different PR